### PR TITLE
apps/testcase/media : Add ITC for media

### PR DESCRIPTION
**Pull request by vir2u3**
Developed test case below
2 testcases in  itc_media_fileinputdatasource.cpp
Check the validation for getChannel and getSampleRate for fileinputdatasource
2 testcase in itc_media_fileoutputdatasource.cpp
Check the validation for getChannel and getSampleRate for fileoutputdatasource
2 testcases in itc_media_mediaplayer.cpp
Create media player and destroy media player
2 testcases in itc_media_mediarecorder.cpp
Create media recorder and destroy media recorder

Signed-off-by: sri <sri@s.sa.corp.samsungelectronics.net>

### CODE REVIEW

**Comment by vir2u3** [comment]
@samegautam @manoj11588612 please review

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+#define FILE_DEFAULT_RATE 16000
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+static const char file_path[] = MEDIA_TEST_FILE_PATH;
+static const char test_data[] = "dummydata";
+static unsigned char *buf;
+
+static void SetUp(void)
```
Comment (samegautam) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: void -> int and check the success case value in main call.

File: apps/examples/testcase/ta_tc/media/itc/itc_media_main.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+	}
+
+#ifdef CONFIG_MEDIA_PLAYER
+	itc_media_MediaPlayer_main();
+	itc_media_FileInputDataSource_main();
+#endif
+#ifdef CONFIG_MEDIA_RECORDER
+	itc_media_mediarecorder_main();
+	itc_media_fileoutputdatasource_main();
+	unlink("/ramfs/record");
```
Comment (samegautam) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_main.cpp]: Please clarify line : unlink("/ramfs/record");

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+{
+	FILE *fp = fopen(file_path, "w");
+	fputs(test_data, fp);
+	fclose(fp);
+	buf = new unsigned char[21];
```
Comment (samegautam) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: Please use of "buf" or it unused.

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+ * Definitions
+ ****************************************************************************/
+#define MEDIA_TEST_FILE_PATH "/mnt/fileinputdatasource.raw";
+#define FILE_DEFAULT_RATE 16000
+
+/****************************************************************************
+ * Global Variables
+ ****************************************************************************/
+static const char file_path[] = MEDIA_TEST_FILE_PATH;
+static const char test_data[] = "dummydata";
```
Comment (samegautam) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: seems like unused variable.

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+ ****************************************************************************/
+
+#include <memory>
+#include <media/MediaPlayer.h>
+#include <media/FileInputDataSource.h>
+#include "tc_common.h"
+
+static const char *file_path = "/mnt/fileinputdatasource.raw";
+
+static void SetUp(void)
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: used  a global flag

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+{
+	FILE *fp = fopen(file_path, "w");
+	fputs("dummydata", fp);
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: done

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+	fclose(fp);
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: done

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+	media::MediaPlayer mp;
+	TC_ASSERT_EQ("create", mp.create(), media::PLAYER_OK);
+	TC_ASSERT_EQ("recreate", mp.create(), media::PLAYER_ERROR);
+	TC_ASSERT_EQ("destroy", mp.destroy(), media::PLAYER_OK);
+	TC_SUCCESS_RESULT();
+}
+
+int itc_media_MediaPlayer_main(void)
+{
+	SetUp();
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: done with global flag

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+* @scenario         create and destroy media recorder and check its validation
+* @apicovered       create,destroy
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_media_MediaRecorder_create_n(void)
+{
+	MediaRecorder media_recorder;
+	TC_ASSERT_EQ("create", media_recorder.create(), RECORDER_OK);
+	TC_ASSERT_EQ("create", media_recorder.create(), RECORDER_ERROR);
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp]: done

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp (hunk 1 above)
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: used a global flag

File: apps/examples/testcase/ta_tc/media/itc/itc_media_main.cpp (hunk 2 above)
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_main.cpp]: done

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp (hunk 3 above)
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: removed

**Comment by seinfra** [comment]
Target : [4e0d5be40b928cd7dfa511c0bb277ff5f00b8ad3] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [4e0d5be40b928cd7dfa511c0bb277ff5f00b8ad3] - Code Rule Check (C++) OK.

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (thapav) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: same as above
Please test dummydata written to file_path using TC_ASSERT_EQ macro

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+{
+	MediaRecorder media_recorder;
+	TC_ASSERT_EQ("create", media_recorder.create(), RECORDER_OK);
+	TC_ASSERT_EQ("destroy", media_recorder.destroy(), RECORDER_OK);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase         itc_media_MediaRecorder_create_n
+* @brief            create media player two times
```
Comment (thapav) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp]: n times*

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+* @scenario         create and destroy media recorder and check its validation
+* @apicovered       create,destroy
+* @precondition     NA
+* @postcondition    NA
+*/
+static void itc_media_MediaRecorder_create_n(void)
+{
+	MediaRecorder media_recorder;
+	TC_ASSERT_EQ("create", media_recorder.create(), RECORDER_OK);
+	for(int i=0; i<10; i++){
```
Comment (thapav) [inline apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp]: please add appropriate macro for 10

File: apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp [hunk 12]
```diff
@@ -0,0 +1,90 @@
...
+static const char *file_path = MEDIA_TEST_FILE_PATH;
+static const char *test_data = "dummydata";
+static int g_flag=0;
+
+static void SetUp(void)
+{
+	FILE *fp;
+	fp = fopen(file_path, "w");
+	TC_ASSERT_NEQ("fopen", fp, NULL);
+	TC_ASSERT_NEQ_CLEANUP("fputs", fputs(test_data, fp), EOF, fclose(fp));
```
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_fileinputdatasource.cpp]: done

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp (hunk 10 above)
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp]: it is negative testcase

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp (hunk 11 above)
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediarecorder.cpp]: done

**Comment by seinfra** [comment]
Target : [aaabddfeea2b0c4b0b73e5d4ca8f207100593d49] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [aaabddfeea2b0c4b0b73e5d4ca8f207100593d49] - Code Rule Check (C++) OK.

File: apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp (hunk 8 above)
[the remark this answers is not in the document]
Comment (vir2u3) [reply apps/examples/testcase/ta_tc/media/itc/itc_media_mediaplayer.cpp]: done

**Comment by vir2u3** [comment]
@thapav @junmin-kim @Taejun-Kwon please review

**Pull request merged**